corehq/apps/translations/integrations/transifex/client.py

```diff
@@ -170,18 +138,16 @@ def get_translation(self, resource_slug, hq_lang_code, lock_resource):
         :param lock_resource: lock resource after pulling translation
         :return: list of POEntry objects
         """
-        lang = self.transifex_lang_code(hq_lang_code)
-        url = "https://www.transifex.com/api/2/project/{}/resource/{}/translation/{}/?file".format(
-            self.project, resource_slug, lang
-        )
-        response = requests.get(url, auth=self._auth, stream=True)
-        if response.status_code != 200:
-            raise ResourceMissing
+        language_id = self._lang_code_to_language_id(self.transifex_lang_code(hq_lang_code))
+        language = self.api.Language(id=language_id)
+        resource = self.api.Resource.get(slug=resource_slug, project=self.project)
+        download = self.api.ResourceTranslationsAsyncDownload.download(resource=resource, language=language)
+        response = requests.get(download, stream=True)
         temp_file = tempfile.NamedTemporaryFile()
         with open(temp_file.name, 'w', encoding='utf-8') as f:
-            f.write(response.content.decode(encoding='utf-8'))
+            f.write(response.content.decode(encoding="utf-8"))
         if lock_resource:
-            self.lock_resource(resource_slug)
+            resource.save(accept_translations=False)
```

Comment:
nit: why not update the "lock_resource" function and retain it?

Reply:
This is a good call, particularly since I'm wrapping _every other_ API call except this one. Will update.